### ISSUE TITLE
Adapt "Contributing" document for non-official library contexts

### ISSUE THREAD

**Issue by EpicEric** [ponylang/library-project-starter#12]
The current `CONTRIBUTING.md` file has been mostly ported from the official repos, making it a lot more relevant for "official Pony code" than for an unofficial user-created library. Here are a few suggestions for changes to make it more useful for user code:

- [ ] **Change the "Bug report" section to "Submitting issues"**. The CONTRIBUTING document in ponyc breaks "bug reports" and "changes/features" into two things because of the RFC process. Obviously user code doesn't have to go through this process, so this section should be expanded to incorporate anything that isn't a bug report. [Here's a good starting point if you're lost](https://opensource.guide/how-to-contribute/#how-to-submit-a-contribution).
- [ ] **Remove reference to official "Slack"** as it's not necessarily a place that the maintainer might actually drop by.
- [ ] **Add section regarding licensing**, since a library can opt for a different license than the BSD-2-Clause for official code. Something along the lines of "refer to the license of this library before making contributions to see any restrictions on source code availability for forks" would make sense.

It'd make sense to break these changes into several PRs instead of one, which is why I set up a task list for this issue.

**Comment by SeanTAllen** [comment]
Re: Github/Gitlab. A giant chunk of the code to automate things in here is GitHub specific so I think leaving GitHub specific language and noting in USAGE.md makes sense (and FYI, I think I'm going to move USAGE.md content to a wiki as it would be easier to maintain going forward)

**Comment by EpicEric** [comment]
Good point. I've moved that part to issue #13, since it doesn't directly affect `CONTRIBUTING.md`.
